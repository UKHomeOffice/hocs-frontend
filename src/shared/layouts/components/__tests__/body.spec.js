import React from "react";
import Body from "../body.jsx";

describe('Layout body component', () => {
    it('should render with default props', () => {
        expect(
            render(<Body/>)
        ).toMatchSnapshot();
    });
    it('should render with phase banner when passed', () => {
        const props = {
            phaseBanner: {
                isVisible: true
            }
        };
        expect(
            render(<Body {...props}/>)
        ).toMatchSnapshot();
    });
});