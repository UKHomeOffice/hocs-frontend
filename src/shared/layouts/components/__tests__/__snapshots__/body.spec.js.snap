// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout body component should render with default props 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="govuk-width-container"
      >
        <main
          class="govuk-main-wrapper"
          id="main-content"
          role="main"
          tabindex="-1"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="govuk-width-container"
    >
      <main
        class="govuk-main-wrapper"
        id="main-content"
        role="main"
        tabindex="-1"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Layout body component should render with phase banner when passed 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="govuk-width-container"
      >
        <div
          class="govuk-phase-banner"
        >
          <p
            class="govuk-phase-banner__content"
          >
            <strong
              class="govuk-tag govuk-phase-banner__content__tag"
            >
              PROTOTYPE
            </strong>
            <span>
              This is a new service – your 
              <a
                href="/"
              >
                feedback
              </a>
               will help us to improve it.
            </span>
          </p>
        </div>
        <main
          class="govuk-main-wrapper"
          id="main-content"
          role="main"
          tabindex="-1"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="govuk-width-container"
    >
      <div
        class="govuk-phase-banner"
      >
        <p
          class="govuk-phase-banner__content"
        >
          <strong
            class="govuk-tag govuk-phase-banner__content__tag"
          >
            PROTOTYPE
          </strong>
          <span>
            This is a new service – your 
            <a
              href="/"
            >
              feedback
            </a>
             will help us to improve it.
          </span>
        </p>
      </div>
      <main
        class="govuk-main-wrapper"
        id="main-content"
        role="main"
        tabindex="-1"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
